sopel/modules/remind.py

```diff
@@ -163,50 +163,103 @@ def remind(bot, trigger):
+                       r'(?:\s+(\d*)\W*(\d*)\W*(\d*))?'     # Big capture group to match date components, up to three
+                       r'(.*)')                             # Finally, match the message
     match = regex.match(trigger.group(2))
     if not match:
         bot.reply("Sorry, but I didn't understand your input.")
         return NOLIMIT
-    hour, minute, second, tz, message = match.groups()
-    if not second:
-        second = '0'
+    hour, minute, seconds, tz, first, second, third, message = match.groups()
+    if pytz:
+        pytz_timezone = pytz.timezone(get_timezone(bot.db, bot.config, tz,
+                                                   trigger.nick, trigger.sender))
+        now = datetime.now(pytz_timezone)
+    else:
+        now = datetime.now()
+
+    year = now.year
+    month = now.month
+    day = now.day
+
+    if not seconds:
+        seconds = '0'
+    if first and second and not third:  # Only two groups - parse as month and day
+        first = int(first)
+        second = int(second)
+
+        if first > 12 and second > 12:
+            bot.reply("Sorry, I didn't understand the date you gave. Try yyyy-mm-dd.")
+            return NOLIMIT
+        elif first <= 12 and second > 12:
+            day = first
+            month = second
+        else:   # Default is to assume that the user gave mm/dd
+            month = first
+            day = second
+    elif third:
+        first = int(first)
+        second = int(second)
+        third = int(third)
+
+        if third > 31:
+            year = third
+            if first > 12:   # If first is probably day
+                day = first
+                month = second
+            else:
+                month = first
+                day = second
+        else:   # I assume that no country uses yyyy-dd-mm.
+            year = first
+            month = second
+            day = third
 
     if pytz:
         timezone = get_timezone(bot.db, bot.config, tz,
                                 trigger.nick, trigger.sender)
         if not timezone:
             timezone = 'UTC'
-        now = datetime.now(pytz.timezone(timezone))
-        at_time = datetime(now.year, now.month, now.day,
-                           int(hour), int(minute), int(second),
-                           tzinfo=now.tzinfo)
-        timediff = at_time - now
+
+        pytz_timezone = pytz.timezone(timezone)
+        now = pytz_timezone.localize(datetime.now())
+        at_time = datetime(year, month, day,
+                           int(hour), int(minute), int(second))
+        at_time = pytz_timezone.localize(at_time)
+        timediff = pytz_timezone.normalize(at_time) - now
     else:
         if tz and tz.upper() != 'UTC':
-            bot.reply("I don't have timezone support installed.")
+            bot.reply("I don't have timzeone support installed.")
```

Comment:
This change introduces a spelling error that wasn't there before. Lol how? :joy_cat:

Reply:
I removed that code in my own copy then put it back in for the PR back then, evidently. How strange.